crates/sparrow-runtime/src/prepare/column_behavior.rs

```diff
@@ -0,0 +1,525 @@
+use anyhow::Context;
+use arrow::{
+    array::ArrayRef,
+    datatypes::{DataType, Field, SchemaRef},
+    record_batch::RecordBatch,
+};
+use sparrow_core::context_code;
+
+use super::PrepareMetadata;
+use std::{borrow::BorrowMut, sync::Arc};
+
+use anyhow::anyhow;
+use arrow::array::{Array, UInt64Array};
+
+use error_stack::{IntoReport, IntoReportCompat, ResultExt};
+use sparrow_core::utils::make_null_array;
+use sparrow_kernels::order_preserving_cast_to_u64;
+
+use crate::{execute::key_hash_inverse::ThreadSafeKeyHashInverse, prepare::Error};
+
+/// Defines how each column in the resulting prepared batch
+/// is computed.
+#[derive(Debug)]
+pub enum ColumnBehavior {
+    /// Cast the given column to the given data type.
+    Cast {
+        index: usize,
+        data_type: DataType,
+        nullable: bool,
+    },
+    /// Perform an "order preserving" cast from a primitive number to u64.
+    OrderPreservingCastToU64 { index: usize, nullable: bool },
+    /// Reference the given column.
+    Reference { index: usize, nullable: bool },
+    /// Hash the given column.
+    ///
+    /// This field is used for preparing the key column during preparation.
+    PrepareEntityKey { index: usize, nullable: bool },
+    /// Hash the given column.
+    ///
+    /// This field is used for preparing the key column during execution.
+    ExecuteEntityKey { index: usize, nullable: bool },
+    /// A random column
```

Comment:
Update this comment while here? Seems to be missing some words.

Reply:
Updated